src/iap/shared/models/carousel.models.ts

```diff
@@ -0,0 +1,9 @@
+import { Type } from '@angular/core';
+
+export interface SlideComponent<T> {
+	data: T;
+}
+
+export class Slide<T> {
+	constructor(public component: Type<SlideComponent<T>>, public data: T) {}
+}
```

Comment:
You can place the information about selected slide in this class and use it somewhere (just for your information, and, if you want, we can discuss some possible use cases.